app/src/store/data.ts

```diff
@@ -85,100 +100,92 @@ export default function useDataStore() {
     grantRounds.value = grantRoundsList as GrantRound[];
-    grantRoundsCLRData.value = grantDataByRound as Record<string, GrantRoundCLR>;
+    grantRoundsDonationToken.value = SUPPORTED_TOKENS_MAPPING[grantRoundDonationTokenAddress] as TokenInfo;
   }
 
   /**
-   * @notice Helper method that fetches metadata for a Grant or GrantRound, and saves the data
-   * to the state as soon as it's received
-   * @param metaPtrs Array of URLs to resolve
-   * @param metadata Name of the store's ref to assign resolve metadata to
-   */
-  async function fetchMetaPtrs(metaPtrs: string[], metadata: Ref) {
-    const newMetadata = metaPtrs
-      .filter((metaPtr) => !metadata.value[metaPtr])
-      .reduce((prev, cur) => {
-        return {
-          ...prev,
-          [cur]: { status: 'pending' },
-        };
-      }, {});
-    // save these pending metadata objects to state
-    metadata.value = { ...metadata.value, ...newMetadata };
-    // resolve metadata via metaPtr and update state
-    void (await Promise.all(
-      Object.keys(newMetadata).map(async (url) => {
-        try {
-          // save each individual ipfs result into storage
-          let data = getStorage('ipfs-' + url);
-          if (!data) {
-            data = await resolveMetaPtr(url);
-            setStorage('ipfs-' + url, data);
-          }
-          metadata.value[url] = { status: 'resolved', ...data };
-        } catch (e) {
-          metadata.value[url] = { status: 'error' };
-          console.error(e);
-        }
-      })
-    ));
-    // trigger a set at the root
-    metadata.value = { ...metadata.value };
-
-    return metadata;
-  }
-
-  /**
-   * @notice Throttle calls to rawPoll()
+   * @notice Throttle calls to rawPoll() and catch any errors
    */
   let timeout: ReturnType<typeof setTimeout> | undefined;
-  function poll(forceRefresh = false) {
+  async function poll(forceRefresh = false) {
     if (!timeout) {
-      rawPoll(forceRefresh);
-      timeout = setTimeout(function () {
-        timeout = undefined;
-      }, 1000);
+      try {
+        await rawPoll(forceRefresh);
+        timeout = setTimeout(function () {
+          timeout = undefined;
+        }, 1000);
+      } catch (e) {
+        console.log('dGrants: Data fetch error - please check your network -- ', e);
+      }
     }
   }
 
   /**
    * @notice Call this method to poll now, then poll on each new block
    */
-  function startPolling() {
+  async function startPolling() {
     provider.value.removeAllListeners(); // remove all existing listeners to avoid duplicate polling
     provider.value.on('block', (/* block: number */) => void poll());
+    // record the network value to detect changes
+    const networkValue = (await getStorageKey('network'))?.data || network.value;
+    // watch the network
+    watch(
+      () => [network.value],
+      async () => {
+        // clear storage and poll again for all network changes after first load
```

Comment:
Could you clarify what exactly we are clearing here on network changes? Want to make sure we aren't clearing the cart data, because a user might be not connected / connected to a testnet while adding things to their cart, switch to mainnet to checkout, and we don't want them to lose their cart in that case

Reply:
We're clearing anything that goes into localForage, so all cart data should be retained 🙌